### PR TITLE
Add pcre and glib ports

### PR DESCRIPTION
**Pull request by no92**
PCRE should work, as by @Dennisbonke quick grep test.

IIRC, glib needs some outstanding mlibc changes, which will be supplied soon™

### CODE REVIEW

File: patches/glib/0001-Small-fixes.patch [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
++#endif
++#ifndef SO_NREAD
++#define SO_NREAD 0
++#endif
++#ifndef CMSG_DATA
++#define CMSG_DATA(x) ((unsigned char*) x)
++#define CMSG_FIRSTHDR(x) ((struct cmsghdr *) NULL)
++#define CMSG_NXTHDR(x, y) ((struct cmsghdr *) NULL)
++#endif
++#endif
```
Comment (avdgrinten) [inline patches/glib/0001-Small-fixes.patch:69]: These can probably be (mostly?) fixed in mlibc. Maybe that's something @Dennisbonke can integrate into his mlibc PR?

File: patches/glib/0001-Small-fixes.patch [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  * and bionic/libc/private/arpa_nameser.h */
+ typedef struct {
+@@ -924,7 +924,10 @@ free_records (GList *records)
+ }
+ 
+ #if defined(G_OS_UNIX)
+-#ifdef __BIONIC__
++#if defined(__BIONIC__) || defined(__managarm__)
++#if defined(__managarm__)
++static int h_errno;
```
Comment (avdgrinten) [inline patches/glib/0001-Small-fixes.patch:94]: What's `h_errno`?

Comment (no92) [reply patches/glib/0001-Small-fixes.patch:94]: Similar to errno, but for `gethostbyname()` and `gethostbyaddr()`. They are marked as obsolete, so mlibc implementation might not even be needed.

For reference, see the [man page on h_errno](https://linux.die.net/man/3/h_errno).

**Comment by avdgrinten** [comment]
Closed in favor of #23 (which is based on this patch). Thanks a lot for this work!

**Pull request closed without merging**